documentation/adoc/cluster-controller.adoc

```diff
@@ -241,6 +266,97 @@ configuration file. For this reason, you can find more information on how to use
+Optional with no default.
+`limits/cpu`::
+the cpu request for the container, corresponding directly to Kubernetes' {resources-link}[`spec.containers[\].resources.limits.cpu`] setting.
+Optional with no default.
+
+[[jvm_json_config]]
+===== JVM Options
+
+It is possible to configure a subset of available JVM options on Kafka, Zookeeper and Topic Controller containers using a JSON object.
+The object has a property for each JVM (`java`) option which can be configured:
+
+`-Xmx`::
+The maximum heap size. See the <<setting_xmx>> section for further details.
+
+`-Xms`::
+The initial heap size.
+Setting the same value for initial and maximum (`-Xmx`) heap sizes avoids the JVM having to allocate memory after startup, at the cost of possibly allocating more heap than is really needed.
+For Kafka and Zookeeper pods such allocation could cause unwanted latency.
+For Kafka Connect avoiding over allocation may be the more important concern, especially in distributed mode where the effects of over-allocation will be multiplied by the number of consumers.
+
+NOTE: The units accepted by JVM settings such as `-Xmx` and `-Xms` are those accepted by the OpenJDK `java` binary in the corresponding image.
+Accordingly, `1g` or `1G` means 1,073,741,824 bytes.
+This is in contrast to the units used for <<resources_json_config,memory limits and requests>>, which follow the Kubernetes convention where `1G` means 1,000,000,000 bytes, and `1Gi` means 1,073,741,824 bytes
+
+.Example Resource limits and requests JSON configuration
+[source,json]
+----
+{
+  "-Xmx": "2g",
+  "-Xms": "2g"
+}
+----
+
+In the above example, the JVM will use 2 GiB (=2,147,483,648 bytes) for its heap.
+Its total memory usage will be approximately 8GiB.
+
+[[setting_xmx]]
+====== Setting `-Xmx`
+
+The default value used for `-Xmx` depends on whether there is a <<resources_json_config,memory request>> for the container:
+
+* If there is a memory request, the JVM's maximum memory will be limited according to the kind of pod (Kafka, Zookeeper, TopicController) to an appropriate value less than the request.
+* Otherwise, when there is no memory request, the JVM's maximum memory will be limited according to the kind of pod and the RAM available to the container.
+
+[IMPORTANT]
+====
+Setting `-Xmx` explicitly is requires some care:
+
+* The JVM's overall memory usage will be approximately 4 × the maximum heap as configured by `-Xmx`.
+
+* If `-Xmx` is set without also setting an appropriate Kubernetes
+memory request, it is possible that the container will be killed should the Kubernetes node
+experience memory pressure (from other Pods running on it).
+====
+
+When `-Xmx` is set explicitly, it is recommended use a memory request that is at least 4.5 × the request heap memory when setting `-Xmx`.
+
+Furthermore, containers doing lots of disk I/O (such as Kafka broker containers) will need to leave some memory available for use as OS page cache.
```

Comment:
we are running on OpenShift, we are saying "page cache" so maybe it's clear that OS is not OpenShift but Operating System but in my opinion it's better to say it explicitly